### PR TITLE
Replace geoID with geometryId consistently

### LINKED ISSUE #393: Inconsistent naming of GeometryID retrieval
In current master, we have two different naming for `Acts::GeometryID` retrieval.

  * `geoID()` in the Core module (127 occurrences)
  * `geometryId()` in the Examples and Fatras modules (19 occurrences)

I would suggest to streamline this.

### PR DESCRIPTION
**Pull request by paulgessinger**
This replaces all occurrences of `geoID()` with `geometryId()`. Also changes a few method names which this naming was part of. All the remaining occurrences of `geoID` or `geoId` should be variable names, which are think are fine to ignore.

Closes #393

### CODE REVIEW

**Comment by asalzburger** [comment]
Ok, I will bin `asalzburger:harmonize-geometryID-retrieval` then, you were faster :-P

File: Core/include/Acts/Geometry/GeometryObject.hpp [hunk 1]
```diff
@@ -35,21 +35,21 @@ class GeometryObject {
 
   /// Constructor from a value
   ///
-  /// @param geoID the geometry identifier of the object
-  GeometryObject(const GeometryID& geoID) : m_geoID(geoID) {}
+  /// @param geometryId the geometry identifier of the object
+  GeometryObject(const GeometryID& geometryId) : m_geometryId(geometryId) {}
```
Comment (asalzburger) [inline Core/include/Acts/Geometry/GeometryObject.hpp:39]: Did you do a global replacement? 

I replaced only the method calls ...

File: Core/include/Acts/Geometry/GeometryObject.hpp [hunk 2]
```diff
@@ -71,19 +71,19 @@ class GeometryObject {
 
   /// Set the value
   ///
-  /// @param geoID the geometry identifier to be assigned
-  void assignGeoID(const GeometryID& geoID);
+  /// @param geometryId the geometry identifier to be assigned
+  void assigngeometryId(const GeometryID& geometryId);
```
Comment (asalzburger) [inline Core/include/Acts/Geometry/GeometryObject.hpp:75]: I think this should then be named `assignGeometryId(...)`

File: Core/src/Geometry/Layer.cpp [hunk 3]
```diff
@@ -48,7 +48,7 @@ Acts::ApproachDescriptor* Acts::Layer::approachDescriptor() {
 void Acts::Layer::closeGeometry(const IMaterialDecorator* materialDecorator,
                                 const GeometryID& layerID) {
   // set the volumeID of this
-  assignGeoID(layerID);
+  assignGeometryId(layerID);
```
Comment (asalzburger) [inline Core/src/Geometry/Layer.cpp:51]: Looks good.

File: Plugins/Json/include/Acts/Plugins/Json/JsonGeometryConverter.hpp [hunk 4]
```diff
@@ -123,9 +123,9 @@ class JsonGeometryConverter {
     /// The data key
     std::string datakey = "data";
     /// The geoid key
-    std::string geoidkey = "Geoid";
+    std::string geometryidkey = "Geoid";
     /// The surface geoid key
-    std::string surfacegeoidkey = "SGeoid";
+    std::string surfacegeometryidkey = "SGeoid";
```
Comment (asalzburger) [inline Plugins/Json/include/Acts/Plugins/Json/JsonGeometryConverter.hpp:128]: I know we hadn't really CamelCased this one ... should be anyway part of a different PR if we decide to do so.

File: Tests/UnitTests/Core/Geometry/TrackingGeometryGeoIDTests.cpp [hunk 5]
```diff
@@ -49,19 +49,19 @@ auto iVolume = constructCylinderVolume(
     "InnerVolume");
```
Comment (asalzburger) [inline Tests/UnitTests/Core/Geometry/TrackingGeometryGeoIDTests.cpp:50]: Consistently the file should then be named `TrackingGeometryGeometryIdTests.cpp` I suppose (and so the UnitTest)

**Review by asalzburger — CHANGES_REQUESTED** [review]
View minor changes.

File: Core/include/Acts/Geometry/GeometryObject.hpp (hunk 2 above)
Comment (asalzburger) [reply Core/include/Acts/Geometry/GeometryObject.hpp:75]: See below, otherwise It would not compile, I fear.

**Comment by msmk0** [comment]
I assigned this to `next` since I think that this does not need to go into `v0.32.0`

File: Core/include/Acts/Geometry/GeometryObject.hpp (hunk 2 above)
Comment (paulgessinger) [reply Core/include/Acts/Geometry/GeometryObject.hpp:75]: Fixing this.

File: Core/include/Acts/Geometry/GeometryObject.hpp (hunk 1 above)
Comment (paulgessinger) [reply Core/include/Acts/Geometry/GeometryObject.hpp:39]: yup

File: Plugins/Json/include/Acts/Plugins/Json/JsonGeometryConverter.hpp (hunk 4 above)
Comment (paulgessinger) [reply Plugins/Json/include/Acts/Plugins/Json/JsonGeometryConverter.hpp:128]: Yeah right, I only touched the spelling, but this should eventually also be changed to camelCase I think (/cc @Corentin-Allaire)

File: Tests/UnitTests/Core/Geometry/TrackingGeometryGeoIDTests.cpp (hunk 5 above)
Comment (paulgessinger) [reply Tests/UnitTests/Core/Geometry/TrackingGeometryGeoIDTests.cpp:50]: Sure, I'll change it.

File: Plugins/Json/include/Acts/Plugins/Json/JsonGeometryConverter.hpp (hunk 4 above)
Comment (Corentin-Allaire) [reply Plugins/Json/include/Acts/Plugins/Json/JsonGeometryConverter.hpp:128]: Yeah, the key used for the json map don't use CamelCased. Not sure why to be honest but I could definitely open a PR to change that. Not really sure we need to replace surfacegeoidkey by surfacegeometryidkey but since you have already done it I guess that is fine.

**Review by asalzburger — APPROVED** [review]

**Comment by asalzburger** [comment]
I am happy now, if the CI builds, I think we should already take this into v0.32.00
